src/google/schema.js

```diff
@@ -32,16 +33,16 @@ const schema = {
   server_datacenter: '%{server.datacenter}V',
   server_region: '%{server.region}V',
   req_http_host: '%v',
-  req_http_X_Host: '%{req.http.X-Orig-Host}V',
-  req_url: 'https://%{req.http.X-Orig-Host}V%{req.http.X-Orig-URL}V',
-  req_http_X_URL: '%{req.http.X-Orig-URL}V',
-  req_http_X_CDN_Request_ID: '%{req.http.X-CDN-Request-ID}V',
+  req_http_X_Host: '%{if(req.http.X-Orig-Host, req.http.X-Orig-Host, req.http.Host)}V',
+  req_url: 'https://%{if(req.http.X-Orig-Host, req.http.X-Orig-Host, req.http.Host)}V%{if(req.http.X-Orig-URL, req.http.X-Orig-URL, req.url)}V',
+  req_http_X_URL: '%{if(req.http.X-Orig-URL, req.http.X-Orig-URL, req.url)}V',
+  req_http_X_CDN_Request_ID: '%{if(req.http.X-CDN-Request-ID, req.http.X-CDN-Request-ID, randomstr(8, "0123456789abcdef") + "-" + randomstr(4, "0123456789abcdef") + "-" + randomstr(4, "0123456789abcdef") + "-" + randomstr(1, "89ab") + randomstr(3, "0123456789abcdef") + "-" + randomstr(12, "0123456789abcdef"))}V',
```

Comment:
What's the point in generating a random ID just for the log? It doesn't serve a purpose and is misleading.